hypothesis-python/tests/numpy/test_gen_data.py

```diff
@@ -1212,3 +1212,44 @@ def test_basic_indices_generate_valid_indexers(
 def test_array_owns_memory(x: np.ndarray):
     assert x.base is None
     assert x[...].base is x
+
+
+@given(nps.arrays(nps.floating_dtypes(), nps.array_shapes()))
+def test_make_noncontiguous(array):
+    np.testing.assert_array_equal(array, nps.make_noncontiguous(array))
+
+
+@given(nps.arrays(nps.floating_dtypes(), nps.array_shapes()), st.data())
+def test_permute_dimensions(array, data):
+    permutation = data.draw(st.permutations(range(array.ndim)))
+    np.testing.assert_array_equal(array, nps.permute_dimensions(array, permutation))
+
+
+@given(nps.arrays(nps.floating_dtypes(), nps.array_shapes()), st.data())
+def test_permute_dimensions_explicit(array, data):
+    perm = data.draw(st.permutations(range(array.ndim)))
+    np.testing.assert_array_equal(
+        array, nps.permute_dimensions(array, permuted_indices=perm)
+    )
+
+
+@given(
+    nps.array_memory_scramblers(), nps.arrays(nps.floating_dtypes(), nps.array_shapes())
+)
+def test_array_memory_scramblers(scrambler, array):
+    scrambled = scrambler(array)
+    # A scrambler should always return an equivalent array
+    np.testing.assert_array_equal(array, scrambled)
+
+
+@settings(deadline=None, max_examples=10)
+@given(nps.arrays(float, nps.array_shapes(min_dims=3, min_side=3)))
```

Comment:
We should be able to handle 1D+ arrays, right?

```suggestion
@given(nps.arrays(float, nps.array_shapes(min_dims=1, min_side=3)))
```

Reply:
Not that also passes `find_any(nps.array_memory_scramblers(), lambda f: strides_wonky(f(array)))`

for that you need enough dimensions to misorder them. I could split this test in two though if you want to test that we can make dim=1 arrays noncontiguous